pkg/server/endpoints/config.go

```diff
@@ -41,16 +50,98 @@ type Config struct {
 	// Bundle endpoint configuration
 	BundleEndpoint bundle.EndpointConfig
 
+	// EnableExperimentalAPI enables the experimental API
+	EnableExperimentalAPI bool
+
 	// CA Manager
 	Manager *ca.Manager
 
 	Log     logrus.FieldLogger
 	Metrics telemetry.Metrics
 }
 
-// New creates new endpoints struct
-func New(c *Config) *Endpoints {
-	return &Endpoints{
-		c: c,
+func (c *Config) makeRegistrationHandler() *registration.Handler {
+	return &registration.Handler{
+		Log:         c.Log.WithField(telemetry.SubsystemName, telemetry.RegistrationAPI),
+		Metrics:     c.Metrics,
+		Catalog:     c.Catalog,
+		TrustDomain: *c.TrustDomain.ID().URL(),
+		ServerCA:    c.ServerCA,
+	}
+}
+
+func (c *Config) makeNodeHandler() (*node.Handler, error) {
+	return node.NewHandler(node.HandlerConfig{
+		Log:                         c.Log.WithField(telemetry.SubsystemName, telemetry.NodeAPI),
+		Metrics:                     c.Metrics,
+		Catalog:                     c.Catalog,
+		TrustDomain:                 *c.TrustDomain.ID().URL(),
+		ServerCA:                    c.ServerCA,
+		Manager:                     c.Manager,
+		AllowAgentlessNodeAttestors: c.AllowAgentlessNodeAttestors,
+	})
+}
+
+func (c *Config) maybeMakeBundleServer() Server {
+	if c.BundleEndpoint.Address == nil {
+		return nil
+	}
+	c.Log.WithField("addr", c.BundleEndpoint.Address).Info("Serving bundle endpoint")
+
+	var serverAuth bundle.ServerAuth
+	if c.BundleEndpoint.ACME != nil {
+		serverAuth = bundle.ACMEAuth(c.Log.WithField(telemetry.SubsystemName, "bundle_acme"), c.Catalog.GetKeyManager(), *c.BundleEndpoint.ACME)
+	} else {
+		serverAuth = bundle.SPIFFEAuth(func() ([]*x509.Certificate, crypto.PrivateKey, error) {
+			state := c.SVIDObserver.State()
+			return state.SVID, state.Key, nil
+		})
+	}
+
+	ds := c.Catalog.GetDataStore()
+	return bundle.NewServer(bundle.ServerConfig{
+		Log:     c.Log.WithField(telemetry.SubsystemName, "bundle_endpoint"),
+		Address: c.BundleEndpoint.Address.String(),
+		Getter: bundle.GetterFunc(func(ctx context.Context) (*bundleutil.Bundle, error) {
+			resp, err := ds.FetchBundle(ctx, &datastore.FetchBundleRequest{
+				TrustDomainId: c.TrustDomain.String(),
```

Comment:
I think that IDString() should be used here.
```suggestion
				TrustDomainId: c.TrustDomain.IDString(),
```

Reply:
Whoops. Yes. Good catch! I've audited all use of `TrustDomain.String()`.